inference-contract/test/guardian_frontend.sh

```diff
@@ -0,0 +1,138 @@
+
+F_SCRIPT=$(basename ${BASH_SOURCE[-1]} )
+F_SERVICE_HOST=${PDO_HOSTNAME}
+F_GUARDIAN_HOST=${PDO_HOSTNAME}
+F_LEDGER_URL=${PDO_LEDGER_URL}
+F_LOGLEVEL=${PDO_LOG_LEVEL:-info}
+F_LOGFILE=${PDO_LOG_FILE:-__screen__}
+F_CONTEXT_FILE=${SOURCE_ROOT}/test/test_context.toml
+F_CONTEXT_TEMPLATES=${PDO_HOME}/contracts/inference/context
+F_EXCHANGE_TEMPLATES=${PDO_HOME}/contracts/exchange/context
+
+F_USAGE='--host service-host | --ledger url | --loglevel [debug|info|warn] | --logfile file'
+SHORT_OPTS='h:l:'
+LONG_OPTS='host:,ledger:,loglevel:,logfile:'
+
+TEMP=$(getopt -o ${SHORT_OPTS} --long ${LONG_OPTS} -n "${F_SCRIPT}" -- "$@")
+if [ $? != 0 ] ; then echo "Usage: ${F_SCRIPT} ${F_USAGE}" >&2 ; exit 1 ; fi
+
+eval set -- "$TEMP"
+while true ; do
+    case "$1" in
+        -h|--host) F_SERVICE_HOST="$2" ; shift 2 ;;
+        -1|--ledger) F_LEDGER_URL="$2" ; shift 2 ;;
+        --loglevel) F_LOGLEVEL="$2" ; shift 2 ;;
+        --logfile) F_LOGFILE="$2" ; shift 2 ;;
+        --help) echo "Usage: ${SCRIPT_NAME} ${F_USAGE}"; exit 0 ;;
+        --) shift ; break ;;
+        *) echo "Internal error!" ; exit 1 ;;
+    esac
+done
+
+F_SERVICE_GROUPS_FILE=${SOURCE_ROOT}/test/${F_SERVICE_HOST}_groups.toml
+F_SERVICE_DB_FILE=${SOURCE_ROOT}/test/${F_SERVICE_HOST}_db
+
+_COMMON_=("--logfile ${F_LOGFILE}" "--loglevel ${F_LOGLEVEL}")
+_COMMON_+=("--ledger ${F_LEDGER_URL}")
+_COMMON_+=("--bind service_host ${F_SERVICE_HOST}")
+_COMMON_+=("--service-groups ${F_SERVICE_GROUPS_FILE}")
+_COMMON_+=("--service-db ${F_SERVICE_DB_FILE}")
+_COMMON_+=("--context-file ${F_CONTEXT_FILE}")
+OPTS=${_COMMON_[@]}
+
+F_KEY_FILES=()
+KEYGEN=${PDO_SOURCE_ROOT}/build/__tools__/make-keys
+
+if [ ! -f ${PDO_HOME}/keys/guardian_service.pem ]; then
+    yell create keys for the guardian service
+    ${KEYGEN} --keyfile ${PDO_HOME}/keys/guardian_service --format pem
+    F_KEY_FILES+=(${PDO_HOME}/keys/guardian_service.pem)
+
+    ${KEYGEN} --keyfile ${PDO_HOME}/keys/guardian_sservice --format pem
+    F_KEY_FILES+=(${PDO_HOME}/keys/guardian_sservice.pem)
+fi
+
+# -----------------------------------------------------------------
+function cleanup {
+    rm -f ${F_SERVICE_GROUPS_FILE}
+    rm -f ${F_SERVICE_DB_FILE} ${F_SERVICE_DB_FILE}-lock
+    rm -f ${F_CONTEXT_FILE}
+    for key_file in ${F_KEY_FILES[@]} ; do
+        rm -f ${key_file}
+    done
+
+    yell "shutdown guardian and storage service"
+    ${PDO_HOME}/contracts/inference/scripts/gs_stop.sh
+    ${PDO_HOME}/contracts/inference/scripts/ss_stop.sh
+}
+
+trap cleanup EXIT
+
+# -----------------------------------------------------------------
+# Start the guardian service and the storage service
+# -----------------------------------------------------------------
+try ${PDO_HOME}/contracts/inference/scripts/ss_start.sh -c -o ${PDO_HOME}/logs -- \
+    --loglevel debug \
+    --config guardian_service.toml \
+    --config-dir ${PDO_HOME}/etc/contracts \
+    --identity guardian_sservice
+
+sleep 3
```

Comment:
what is the purpose of this file?

Reply:
this file is necessary only if we do not use docker for the guardian (to be started before starting the jupyter server). Question is do we even need to support such a mode ? My suggestion is that we always expect the guardian to be deployed via the docker container. Please comment.

Reply:
thanks, have removed this file completely. As part of documentation, have given commands using ss_start and gs_start as to how to deploy guardian directly, if needed for bare-metal.